pokemon_pong.py

```diff
@@ -0,0 +1,168 @@
+        self.ball.centery = window_height / 2.0 #ball starts at center of screen
+        self.ball.centerx = window_width / 2.0
+        self.dx = -22 #setting ball speed
+        self.dy = 5 #setting ball direction
+
+    def set_ball(self): #ball action at beginning of game
+        if self.ball.left > 0 or self.ball.right < window_width:
+            self.ball.move_ip(self.dx, self.dy)
+
+    def collision_wall(self):
+         #instances if ball hits a wall
+        if self.ball.top <= 0: #hits top and changes direction
+            self.dy = -1*self.dy
+        elif self.ball.bottom >= window_height: #hits bottom and changes direction
+            self.dy = -1*self.dy
+        elif self.ball.right >= window_width: #resets ball to center if it hits the left or right sides
+            p1_score()
+            self.ball.center = (600,600)
+            self.dx = -1*self.dx
+        elif self.rect.left <= 0:
+            p2_score()
+            self.ball.center = (600,600)
+            self.dx = -1*self.dx
+
+class Pikachu2(Pikachu):
+    def __init__(self):
+        pygame.sprite.Sprite.__init__(self)
+        self.image = pygame.image.load('pikachu.png').convert_alpha()
+        self.image = pygame.transform.scale(self.image, (150,200)) 
+        self.rect = self.image.get_rect() 
+        self.pikac = self.rect
+        self.pikac.centery = window_height / 2.0 #setting starting location
+        self.pikac.right = window_width
+
+    def move_pika2(self,key):
+        if 150 < self.rect.bottom < window_height:
+            if key[pygame.K_DOWN]:
+                self.rect.centery += 25
+            elif key[pygame.K_UP]:
+                self.rect.centery -= 25
+
+        elif self.rect.bottom >= window_height:
+            self.rect.centery -= 25
+        elif self.rect.bottom <= 150:
+            self.rect.centery += 25
+
+'''Scoring'''
+def p1_score():
+    global p1_wins
+    p1_wins += 1
+    win_game()
+
+def p2_score():
+    global p2_wins
+    p2_wins += 1
+    win_game()
+
+def win_game(): #displays win message and resets score and player positions if someone reaches the score 5
+    global p1_wins, p2_wins
+    if p1_wins == 5: 
+        p1_wins = 0
+        p2_wins = 0
+        p1_wins_text = win_text.render('Player 1 wins!', 1, (255,255,255))
+        screen.blit(p1_wins_text, (300,600))
+        pygame.display.flip()
+        pygame.display.update()
+        time.sleep(2)
+        reset_game()
+    elif p2_wins == 5:
+        p1_wins = 0
+        p2_wins = 0
+        p2_wins_text = win_text.render('Player 2 wins!', 1, (255,255,255))
+        screen.blit(p2_wins_text, (300,600))
+        pygame.display.flip()
+        pygame.display.update()
+        time.sleep(2)
+        reset_game()
+
+def reset_game(): #resets pikachu locations when someone wins the game
+    global pika, pika2 
+    pika.pikac.centery = window_height / 2.0
+    pika2.pikac.centery = window_height / 2.0
+
+'''main'''
```

Comment:
To clarify what the "main" functionality is, I recommend putting this in an if statement -- so, something like
`if __name__ == "__main__":`